solver/src/encoding.rs

```diff
@@ -70,3 +77,71 @@ pub struct EncodedSettlement {
     pub trades: Vec<EncodedTrade>,
     pub interactions: [Vec<EncodedInteraction>; 3],
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn order_flag_permutations() {
+        for (order, flags) in &[
+            (
+                OrderCreation {
+                    kind: OrderKind::Sell,
+                    partially_fillable: false,
+                    sell_token_balance: BalanceFrom::Erc20,
+                    buy_token_balance: BalanceTo::Erc20,
+                    signing_scheme: SigningScheme::Eip712,
+                    ..Default::default()
+                },
+                0b0000000,
+            ),
+            (
+                OrderCreation {
+                    kind: OrderKind::Sell,
+                    partially_fillable: true,
+                    sell_token_balance: BalanceFrom::Erc20,
+                    buy_token_balance: BalanceTo::Internal,
+                    signing_scheme: SigningScheme::Eip712,
+                    ..Default::default()
+                },
+                0b0010010,
```

Comment:
Also, I know you mention in the PR description how these were generated, but maybe we could include a comment on the breakdown of this value.

Reply:
Love the comments!